### DISCUSSION TITLE
How to update content of virtual module

### DISCUSSION THREAD

**Discussion by BOBRIWE** [webpack/webpack#15825] [Q&A]
Hi all.

I'm trying to add a virtual module which exports specific files, which I need to use in bundle.
Content looks like this: 
``` javascript
module.exports = {
    'a-name': import('a.specific.js'),
    'b-name': import('b.specific.js'),
}
```
Also, it should be possible to use it in watch mode, which means the content of this virtual module can be dynamic.

First part of task solved by [webpack-virtual-modules](https://github.com/sysgears/webpack-virtual-modules), but for some reason this plugin updates content only once and new files can't be added to virtual module.

So, maybe someone tried to solve this issue or any suggestions?

Example of my plugin:
``` javascript
const Compilation = require('webpack').Compilation;
const Compiler = require('webpack').Compiler;
const VirtualModulesPlugin = require('webpack-virtual-modules');

const PLUGIN_NAME = 'CustomVirtualPlugin';

class CustomVirtualPlugin {


    /**
     * @param {Compiler} compiler context path for entry
     * @returns {void} returns
     */
    apply(compiler) {
        const virtualModulesPlugin = new VirtualModulesPlugin({'./test-virtual.js': `console.log('Test virtual')`});
        virtualModulesPlugin.apply(compiler);

        let count = 0;

        setInterval(() => {
            console.log('Call rebuild');

            virtualModulesPlugin.writeModule('./test-virtual.js', `console.log('${count++}')`);
        }, 10000)
    }

}

module.exports = CustomVirtualPlugin;
```

Thank you

**Comment by vankop** [comment]
please ask in webpack-virtual-modules issue tracker

**Reply by BOBRIWE** [reply]
Seems like nobody working on this plugin anymore. 
My question is more about how I can implement all of this stuff together from scratch.
I was researching most of webpack factories and module classes, but seems like all of them based on file system (if I need module, file of this module should physically exist on disk). Probably webpack-virtual-modules implemented in hacky way because of that.
I just need normalModuleFactory + NormalModule which can work only with content, not files. Or I missed something?
